server/sonar-server/src/main/java/org/sonar/server/projectlink/ws/SearchAction.java

```diff
@@ -124,8 +124,8 @@ private ComponentDto getComponentByUuidOrKey(DbSession dbSession, SearchWsReques
       request.getProjectKey(),
       ComponentFinder.ParamNames.PROJECT_ID_AND_KEY);
 
-    if (!userSession.hasComponentUuidPermission(UserRole.ADMIN, component.uuid()) &&
-      !userSession.hasComponentUuidPermission(UserRole.USER, component.uuid())) {
+    if (!userSession.hasComponentPermission(UserRole.ADMIN, component) &&
```

Comment:
+1 for `checkComponentAnyPermission(ComponentDto component, String permission, String... otherPermissions)`

I don't have opinion for the comment about param ordering, so let's go in your direction if you prefer it. 

I will do the refactorings in another PR. Thanks.